libcaf_core/caf/detail/parser/read_ipv4_address.hpp

```diff
@@ -53,7 +53,8 @@ void read_ipv4_octet(state<Iterator, Sentinel>& ps, Consumer& consumer) {
   uint8_t res = 0;
   // Reads the a decimal place.
   auto rd_decimal = [&](char c) {
-    return add_ascii<10>(res, c);
+    auto b = add_ascii<10>(res, c) && res <= 255;
```

Comment:
Shouldn't `res <= 255` always be true for `uint8_t`? (And if the check is necessary, shouldn't it use `std::numeric_limits<uint8_t>::max()`.)